dotCMS/src/main/java/com/dotcms/rest/api/v1/content/ContentVersionResource.java

```diff
@@ -165,6 +172,44 @@ public Response findVersions(@Context final HttpServletRequest request,
 
     }
 
+    @GET
+    @JSONP
+    @NoCache
+    @Path("/id/{identifier}/history")
+    @Produces(MediaType.APPLICATION_JSON + ";charset=UTF-8")
+    public Response history(@Context final HttpServletRequest request,
+                            @Context final HttpServletResponse response,
+                            @PathParam("identifier") final String identifier,
+                            @QueryParam("groupByLang") final boolean groupByLanguage,
+                            @QueryParam("limit") int limit,
+                            @QueryParam("offset") int offset)
+            throws DotDataException, DotStateException, DotSecurityException {
+        final InitDataObject initDataObject = new WebResource.InitBuilder(this.webResource)
+                .requestAndResponse(request, response)
+                .rejectWhenNoUser(true)
+                .init();
+        final User user = initDataObject.getUser();
+        final Identifier identifierObj = this.getIdentifier(identifier, user);
+        if (null == identifier) {
+            throw new BadRequestException(getFormattedMessage(user.getLocale(), BAD_REQUEST_ERROR_MESSAGE_KEY));
+        }
+        limit = limit > MAX ? MAX : limit <= 0 ? MIN : limit;
```

Comment:
this should be immutable and should assign to a diff variable, the parameters should be final, encapsulate this into a method